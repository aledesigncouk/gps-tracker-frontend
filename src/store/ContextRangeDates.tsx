import React, { createContext, useState, ReactNode } from 'react';

interface RangeDatesContextProps {
    startDate: Date | null;
    endDate: Date | null;
    setStartDate: (date: Date | null) => void;
    setEndDate: (date: Date | null) => void;
}

const RangeDatesContext = createContext<RangeDatesContextProps | undefined>(undefined);

export const RangeDatesContextStore: React.FC<{ children: ReactNode }> = ({ children }) => {
    const [startDate, setStartDate] = useState<Date | null>(null);
    const [endDate, setEndDate] = useState<Date | null>(null);

    return (
        <RangeDatesContext.Provider value={{
            startDate: startDate,
            setStartDate: setStartDate,
            endDate: endDate,
            setEndDate: setEndDate,
        }}>
            {children}
        </RangeDatesContext.Provider>
    );
};

export const useRangeDatesStore = () => {
    const context = React.useContext(RangeDatesContext);
    if (context === undefined) {
        throw new Error('useRangeDatesStore must be used within a RangeDatesProvider');
    }
    return context;
};

export default RangeDatesContextStore;