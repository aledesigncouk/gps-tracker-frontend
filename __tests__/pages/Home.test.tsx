import { act, render, screen, waitFor } from "@testing-library/react";
import Home from "@pages/index";
import { useRangeDatesStore } from '@store/ContextRangeDates';

jest.mock('@store/ContextRangeDates', () => ({
  useRangeDatesStore: jest.fn(),
}));
describe("Home Page", () => {
  const setStartDate = jest.fn();
  const setEndDate = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();

    useRangeDatesStore.mockReturnValue({
      setStartDate,
      setEndDate,
    });
  });

  it("renders the main component", async () => {
    render(<Home />);

    await waitFor(() => {
      const home = screen.getByTestId("main");
      expect(home).toBeInTheDocument();
    })
  });
});

describe('Home component', () => {
  it('match the snapshot', async () => {
    await waitFor(() => {
      const { asFragment } = render(<Home />);
      expect(asFragment()).toMatchSnapshot();
    })
  })
});
