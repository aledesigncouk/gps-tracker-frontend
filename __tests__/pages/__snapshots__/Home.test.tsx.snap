// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-primary min-vh-100 text-white container-fluid"
    data-testid="main"
  >
    <div
      class="pt-2 d-flex align-items-center row"
      data-testid="topbar"
    >
      <div
        class="col"
      >
        <div
          class="hstack gap-3"
        >
          <div
            class=""
          >
            <label
              class=""
            >
              Start Date
            </label>
            <div>
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class=""
                  data-testid="startDate-input"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <label
              class=""
            >
              End Date
            </label>
            <div>
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class=""
                  data-testid="endDate-input"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="ms-auto vstack"
          data-testid="year-selector"
        >
          <label
            class=""
            for="dropdown"
          >
            Choose a year:
          </label>
          <div
            class="dropdown"
          >
            <button
              aria-expanded="false"
              class="dropdown-toggle btn btn-success"
              data-testid="year-dropdown"
              id="dropdown-basic"
              type="button"
            >
              Select year
            </button>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
